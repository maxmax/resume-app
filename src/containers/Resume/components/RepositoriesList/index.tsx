import React from 'react';
import dayjs from 'dayjs';
import { Repository } from '@/services';
import { 
	Box, 
	Grid,
	Typography,
	Divider, 
} from '@/components';

interface RepositoriesListProps {
  repositories?: Repository[];
}

const RepositoriesList: React.FC<RepositoriesListProps> = ({ repositories }) => {
	
	if (!repositories) return null;

	return (
		<Box sx={{ mt: 4, textAlign: 'center' }}>
			<Grid container spacing={2}>
				<Grid item xs={12}>
					<Typography variant="h4">Popular Repositories</Typography>
				</Grid>
				<Grid item xs={12}>
					{repositories.length ? repositories.map((repo) => (
						<Box key={repo.name}>
							<Typography variant="overline" display="block">
								<a href={repo.html_url} target="_blank" rel="noopener noreferrer">{repo.name}</a> 
							</Typography>
							<Typography variant="overline" display="block">
								(Last Updated: {dayjs(repo.updated_at).format('DD-MMM-YYYY h:mm A')})
							</Typography>
							{repo.language &&
								<Typography variant="caption" display="block" gutterBottom>
									(Language: {repo.language})
								</Typography>
							}
							<Divider />
						</Box>
					)) : 'Empty ðŸ¤”'}
				</Grid>
			</Grid>
		</Box>
	);
};

export default RepositoriesList;
