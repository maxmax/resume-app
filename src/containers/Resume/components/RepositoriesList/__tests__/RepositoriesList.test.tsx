import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import RepositoriesList from '..';

// Ваші тестові дані
const mockRepositories = [
  {
    id: 1,
    name: 'Repo 1',
    updated_at: '2023-01-01T00:00:00Z',
    language: 'JavaScript',
    full_name: 'username/repo1',
    html_url: 'https://example.com/repo1',
    languages_url: 'https://example.com/repo1/languages',
  },
  {
    id: 2,
    name: 'Repo 2',
    updated_at: '2023-01-02T00:00:00Z',
    language: 'TypeScript',
    full_name: 'username/repo2',
    html_url: 'https://example.com/repo2',
    languages_url: 'https://example.com/repo2/languages',
  },
];

describe('RepositoriesList component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
			<MemoryRouter>
				<RepositoriesList
					repositories={mockRepositories}
					currentPage={1}
					perPage={5}
					count={mockRepositories.length}
					selectedSort="updated"
					onPageChange={jest.fn()}
					onPerPageChange={jest.fn()}
					onSortChange={jest.fn()}
				/>
			</MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('RepositoriesList component', () => {
  it('renders repositories correctly', () => {
    render(
			<MemoryRouter>
				<RepositoriesList
					repositories={mockRepositories}
					currentPage={1}
					perPage={5}
					count={mockRepositories.length}
					selectedSort="updated"
					onPageChange={jest.fn()}
					onPerPageChange={jest.fn()}
					onSortChange={jest.fn()}
				/>
			</MemoryRouter>
    );

    // Checking that the repositories are displayed correctly
    expect(screen.getByText('Repo 1')).toBeTruthy();
    expect(screen.getByText('Repo 2')).toBeTruthy();

    // Checking if the language of the repositories is displayed
    expect(screen.getByText('JavaScript')).toBeTruthy();
    expect(screen.getByText('TypeScript')).toBeTruthy();

    // Checking if "Show more" buttons are visible
    const showMoreButtons = screen.getAllByText('Show more');
    expect(showMoreButtons).toHaveLength(mockRepositories.length);

    // Checking if pagination is displayed
		expect(screen.getByText('1–2 of 2')).toBeTruthy();
    expect(screen.getByLabelText('Rows per page:')).toBeTruthy();
  });

  it('displays message when there are no repositories', () => {
    render(
      <RepositoriesList
        repositories={[]}
        currentPage={1}
        perPage={5}
        count={0}
        selectedSort="updated"
        onPageChange={jest.fn()}
        onPerPageChange={jest.fn()}
        onSortChange={jest.fn()}
      />
    );

    expect(screen.getByText("Doesn't have any public repositories yet 🤔")).toBeTruthy();
  });
});
