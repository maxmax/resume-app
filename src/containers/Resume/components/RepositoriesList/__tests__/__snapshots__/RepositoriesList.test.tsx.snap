// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoriesList component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-ilk07l"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-1xzqqmo-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-rukeog-MuiGrid-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
        >
          Repositories:
        </h4>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-124q3z-MuiGrid-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-sizeSmall  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-controls=":r0:"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-label="Sort By"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            role="combobox"
            tabindex="0"
          >
            Updated
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value="updated"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
      >
        <table
          aria-label="simple table"
          class="MuiTable-root css-quj9j6-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                <b>
                  Name
                </b>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
                scope="col"
              >
                <b>
                  Updated
                </b>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
                scope="col"
              >
                <b>
                  Language
                </b>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
                scope="col"
              />
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                scope="row"
              >
                Repo 1
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-107r4zp-MuiTableCell-root"
              >
                01-Jan-2023
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-14ju0jl-MuiTableCell-root"
              >
                JavaScript
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-11m86qr-MuiTableCell-root"
              >
                <a
                  href="/resume/username/repo1"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall css-1k23hlb-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Show more
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </a>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                scope="row"
              >
                Repo 2
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-107r4zp-MuiTableCell-root"
              >
                02-Jan-2023
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-14ju0jl-MuiTableCell-root"
              >
                TypeScript
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-11m86qr-MuiTableCell-root"
              >
                <a
                  href="/resume/username/repo2"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall css-1k23hlb-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Show more
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </a>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="MuiTablePagination-root css-jtlhu6-MuiTablePagination-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
        >
          <div
            class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
          />
          <p
            class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
            id=":r2:"
          >
            Rows per page:
          </p>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-input css-16c50h-MuiInputBase-root-MuiTablePagination-select"
          >
            <div
              aria-controls=":r3:"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby=":r2: :r1:"
              class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
              id=":r1:"
              role="combobox"
              tabindex="0"
            >
              5
            </div>
            <input
              aria-hidden="true"
              aria-invalid="false"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value="5"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
          >
            1–2 of 2
          </p>
          <div
            class="MuiTablePagination-actions"
          >
            <button
              aria-label="Go to previous page"
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              title="Go to previous page"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowLeftIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </button>
            <button
              aria-label="Go to next page"
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              title="Go to next page"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowRightIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
  </div>
</DocumentFragment>
`;
